crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -193,6 +194,19 @@ impl Inner {
     }
 }
 
+fn should_cache(result: &Result<f64, PriceEstimationError>) -> bool {
+    // We don't want to cache errors that we consider transient
+    match result {
```

Comment:
suggestion: you can combine all the `false` and `true` patterns into one which could make it easier to read:
```rust
    use PriceEstimationError::*;
    match result {
        Ok(_)
        | Err(NoLiquidity)
        | Err(ZeroAmount)
        | Err(UnsupportedToken { .. })
        | Err(UnsupportedOrderType) => true,
        Err(Other(_)) | Err(RateLimited) => false,
    }
```